fog/view/server/tests/smoke_tests.rs

```diff
@@ -61,8 +52,7 @@ fn get_test_environment(
     let db_test_context = SqlRecoveryDbTestContext::new(logger.clone());
     let db = db_test_context.get_db_instance();
 
-    let port = PORT_NR.fetch_add(1, Ordering::SeqCst) as u16;
-
+    let port = portpicker::pick_unused_port().expect("pick_unused_port");
```

Comment:
Is this really the right thing to use? It seems like there is a time-of-check vs time-of-use issue:

https://docs.rs/portpicker/latest/portpicker/fn.pick_unused_port.html#

That being said, I don't see a great alternative right now. I liked that the atomic port counter thing ensured everyone gets a unique port but making that work across processes would be very complicated. I wonder at what point the unsynchronized port licker starts to fail

Reply:
It looks to me that it's choosing random ports in a range of 10,000 https://docs.rs/portpicker/latest/src/portpicker/lib.rs.html#57

Reply:
Fair point. The number of jobs is controllable, and defaults to the number of cores on the machine. So we should generally be running < 100 tests concurrently.

Reply:
It is also a small window between when the port is picked and when it is bound/used.